substrate/frame/support/src/lib.rs

```diff
@@ -2729,6 +2254,162 @@ pub mod pallet_macros {
 	/// Now, this can be executed as follows:
 	#[doc = docify::embed!("src/tests/tasks.rs", tasks_work)]
 	pub use frame_support_procedural::tasks_experimental;
+
+	/// The optional attribute `#[pallet::whitelist_storage]` will declare the
+	/// storage as whitelisted from benchmarking. Doing so will exclude reads of
+	/// that value's storage key from counting towards weight calculations during
+	/// benchmarking.
+	///
+	/// This attribute should only be attached to storages that are known to be
+	/// read/used in every block. This will result in a more accurate benchmarking weight.
+	///
+	/// ### Example
+	/// ```
+	/// #[frame_support::pallet]
+	/// mod pallet {
+	/// 	use frame_support::pallet_prelude::*;
+	///
+	/// 	#[pallet::pallet]
+	/// 	pub struct Pallet<T>(_);
+	///
+	/// 	#[pallet::storage]
+	/// 	#[pallet::whitelist_storage]
+	/// 	pub type MyStorage<T> = StorageValue<_, u32>;
+	///
+	/// 	#[pallet::config]
+	/// 	pub trait Config: frame_system::Config {}
+	/// }
+	/// ```
+	///
+	/// NOTE: As with all `pallet::*` attributes, this one _must_ be written as
```

Comment:
I think this should be explained once in the docs for `frame_support::pallet` and not here.